pkg/registry/service/rest.go

```diff
@@ -283,6 +283,21 @@ func (rs *REST) ResourceLocation(ctx api.Context, id string) (*url.URL, http.Rou
 		return nil, nil, errors.NewBadRequest(fmt.Sprintf("invalid service request %q", id))
 	}
 
+	// If a port number was specified...
+	if portNum, err := strconv.ParseInt(portStr, 10, 64); err == nil {
+		// look up the service...
+		if svc, err := rs.registry.GetService(ctx, svcName); err == nil {
+			// find the corresponding service port or targetPort...
+			for _, svcPort := range svc.Spec.Ports {
+				if svcPort.Port == int(portNum) {
+					// use the declared port's name
+					portStr = svcPort.Name
```

Comment:
Will this work if there is only one service port and its "Name" field is empty?  I think it might work since in this case, both svcPort.Name and ss.Ports[i].Name will be empty string?